packages/flutter/lib/src/cupertino/page_scaffold.dart

```diff
@@ -84,4 +94,14 @@ class CupertinoPageScaffold extends StatelessWidget {
       ),
     );
   }
-}
\ No newline at end of file
+}
+
+/// Widget that has a preferred size and reports whether it fully obstructs
+/// widgets behind it.
```

Comment:
please add a lot more documentation here. imagine someone has never heard of this widget and they want to know why it exists, why to use it, how to use it (both on the implementing side and the consuming side), etc.